### PR TITLE
[iOS] CollectionView multi-item Add/Remove/Replace/Move

### PR DESCRIPTION
**Pull request by hartez**
### Description of Change ###

Update INotifyCollectionChanged handling for CollectionView on iOS to handle mutli-item Add, Remove, Replace, and Move notifications. 

### Issues Resolved ### 

- partially implements #3172 

### API Changes ###

 None

### Platforms Affected ### 

- iOS

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### Testing Procedure ###

In Control Gallery, navigate to CollectionView Gallery -> Observable Collection Galleries -> [Multi-item add/remove, no index|Multi-item add/remove, with index].

### PR Checklist ###

- [ ] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

**Comment by kingces95** [comment]
So I know this is an iOS change but.. when I open the Android Add/Remove Items (List) and tap `remove` then `insert` I expect the first item to go bye-bye (which it does with a nice animation) and for another item to appear. But actually nothing appears. Then if I hit remove once nothing happens. But if I hit it again the top item is removed. 

Sooo many combinations but this UITest is all teed up for automation so, save us UITest!

**Comment by minaairsupport** [comment]
I got this crash Android , you can reproduce it in an easy way init with empty itemSource then load data from webservice then fill this data source 

```
JniEnvironment+InstanceMethods.CallNonvirtualVoidMethod (Java.Interop.JniObjectReference instance, Java.Interop.JniObjectReference type, Java.Interop.JniMethodInfo method, Java.Interop.JniArgumentValue* args)
Java.Lang.IllegalStateException: Observer md51558244f76c53b6aeda52c8a337f2c37.DataChangeObserver@e23f401 was not registered.
```

**Comment by hartez** [comment]
> I got this crash Android , you can reproduce it in an easy way init with empty itemSource then load data from webservice then fill this data source
> 
> ```
> JniEnvironment+InstanceMethods.CallNonvirtualVoidMethod (Java.Interop.JniObjectReference instance, Java.Interop.JniObjectReference type, Java.Interop.JniMethodInfo method, Java.Interop.JniArgumentValue* args)
> Java.Lang.IllegalStateException: Observer md51558244f76c53b6aeda52c8a337f2c37.DataChangeObserver@e23f401 was not registered.
> ```

@minaairsupport Please open a separate issue for this.

**Comment by hartez** [comment]
@paymicro Ping.

File: Xamarin.Forms.Platform.iOS/CollectionView/ObservableItemsSource.cs [hunk 1]
```diff
@@ -44,18 +48,38 @@ void CollectionChanged(object sender, NotifyCollectionChangedEventArgs args)
...
+			}
+
+			var start = Math.Min(args.OldStartingIndex, args.NewStartingIndex);
+			var end = Math.Max(args.OldStartingIndex, args.NewStartingIndex) + count;
+			_collectionView.ReloadItems(CreateIndexesFrom(start, end));
 		}
 		
 		private void Replace(NotifyCollectionChangedEventArgs args)
 		{
 			var startIndex = args.NewStartingIndex > -1 ? args.NewStartingIndex : _itemsSource.IndexOf(args.NewItems[0]);
```
Comment (paymicro) [inline Xamarin.Forms.Platform.iOS/CollectionView/ObservableItemsSource.cs]: this var need only for `ReloadItems`. Please move it into `if` block below.

Comment (hartez) [reply Xamarin.Forms.Platform.iOS/CollectionView/ObservableItemsSource.cs]: Fixed.

**Comment by hartez** [comment]
> `Remove` 0,3 - removes 3 items (excluded 3)
> `Replace` 0,3 - removes 4 items (included 3)
> `Move` 0,3,5 - removes 4 items
> It is necessary either to include the `end` index everywhere or to exclude it.

Fixed the Remover so it includes the end index.

**Pull request merged**